R/matrix_form.R

```diff
@@ -905,7 +917,98 @@ basic_matrix_form <- function(df, parent_path = "root") {
     nrow_header = 1,
     has_rowlabs = TRUE
   )
-  mform_build_refdf(ret)
+  ret <- mform_build_refdf(ret)
+
+  if (add_decoration) {
+    main_title(ret) <- "main title"
+    main_footer(ret) <- c("main", "  footer")
+    prov_footer(ret) <- "prov footer"
+    subtitles(ret) <- c("sub", "titles")
+  }
+
+  ret
+}
+
+#' Create spoof matrix form from a data.frame
+#'
+#' @describeIn test_matrix_form Create a `MatrixPrintForm` object from a data.frame `df` that
+#'   respects the default formats for a listing object.
+#'
+#' @param keycols character. Vector of `df` column names that are printed first and
+#'   repeated values are assigned to `""`. This format is characteristic of a listing matrix form.
+#'   When `NULL`, no key columns are used. Defaults to `c("vs", "gear")` for `mtcars` default dataset.
+#' @return A valid `MatrixPrintForm` object representing `df` as a listing,
+#'   ready for ASCII rendering.
+#'
+#' @examples
+#' mform <- basic_listing_mf(mtcars)
+#' cat(toString(mform))
+#'
+#' @export
+basic_listing_mf <- function(df,
```

Comment:
hey @Melkiades , what do you think if we move this function to rlistings?

Reply:
This function needs to be here because its purpose is avoiding importing {rlistings} to test listings in {formatters} directly